h/browser/chrome/test/uri-info-test.js

```diff
@@ -0,0 +1,261 @@
+'use strict';
+
+var proxyquire = require('proxyquire');
+
+describe('uriInfo', function() {
+  var server;
+  var uri = 'http://example.com/example';
+  var uriInfo;
+  var serviceUrl = 'http://hypothes.is';
+
+  beforeEach(function() {
+    server = sinon.fakeServer.create({
+      autoRespond: true,
+      respondImmediately: true
+    });
+
+    sinon.stub(console, 'error');
+
+    var settingsPromise = Promise.resolve({serviceUrl: serviceUrl});
+    settingsPromise['@noCallThru'] = true;
+    uriInfo = proxyquire(
+      '../lib/uri-info', {'./settings': settingsPromise});
+  });
+
+  afterEach(function() {
+    server.restore();
+    console.error.restore();
+  });
+
+  it('sends the correct XMLHttpRequest to the server', function() {
+    server.respondWith(
+      'GET', serviceUrl + '/app/uriinfo?uri=' + uri,
+      [200, {}, '{"total": 3, "blocked": false}']
+    );
+
+    return uriInfo(uri).then(
+      function onResolved() {
+        assert(server.requests.length === 1);
+        var request = server.requests[0];
+        assert(request.method === 'GET');
```

Comment:
Use `assert.equal()` rather than `assert(x === y)`. Unlike Python's testing suite, chai won't automatically show the arguments if the assertion fails.